src/main/oAuthClient.ts

```diff
@@ -645,11 +645,16 @@ export default class OAuthClient {
       ...params
     })
     if(auth.useWebMessage) {
-      return this.getWebMessage(
-        `${this.passwordlessVerifyUrl}?${queryString}`,
-        this.config.baseUrl,
-        auth.redirectUri
+      const timeout = (delay: number): Promise<void> => new Promise((resolve) => setTimeout(() => resolve(), delay))
+      const promiseGetWebMessage: Promise<void> = this.getWebMessage(
+          `${this.passwordlessVerifyUrl}?${queryString}`,
+          this.config.baseUrl,
+          auth.redirectUri
       ).then()
+        return Promise.race([
+          promiseGetWebMessage,
+          timeout(1000)
+        ])
```

Comment:
```suggestion
      const promiseGetWebMessage: Promise<void> = this.getWebMessage(
          `${this.passwordlessVerifyUrl}?${queryString}`,
          this.config.baseUrl,
          auth.redirectUri
      )
      return Promise.race([
        promiseGetWebMessage,
        timeout(1000)
      ])
```
Sinon ça veut dire que tu attends que `promiseGetWebMessage` soit traité, donc ta race condition se fera entre la promesse déjà résolue et le timeout, donc ça ne sert à rien...

Reply:
Par conte à voir si le timeout de 1sec n'est pas un peu court.